analyzers/src/SonarAnalyzer.Common/Rules/RouteTemplateShouldNotStartWithSlashBase.cs

```diff
@@ -0,0 +1,107 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Concurrent;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class RouteTemplateShouldNotStartWithSlashBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S6931";
+    private const string MessageOnlyActions = "Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.";
+    private const string MessageActionsAndController = "Change the paths of the actions of this controller to be relative and add a controller route with the common prefix.";
+
+    protected abstract TSyntaxKind MethodSyntaxKind { get; }
+    protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
+
+    protected abstract Location ControllerLocation(INamedTypeSymbol symbol);
+
+    protected override string MessageFormat => "{0}";
+    protected RouteTemplateShouldNotStartWithSlashBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(compilationStartContext =>
+        {
+            if (!ShouldRegisterAction(compilationStartContext.Compilation))
+            {
+                return;
+            }
+
+            compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
+            {
+                var symbol = (INamedTypeSymbol)symbolStartContext.Symbol;
+                if (symbol.IsControllerType())
+                {
+                    var controllerActionInfo = new ConcurrentStack<ControllerActionInfo>();
+                    symbolStartContext.RegisterSyntaxNodeAction(nodeContext =>
+                    {
+                        if (nodeContext.SemanticModel.GetDeclaredSymbol(nodeContext.Node) is IMethodSymbol methodSymbol && methodSymbol.IsControllerMethod())
+                        {
+                            controllerActionInfo.Push(new ControllerActionInfo(methodSymbol, RouteAttributeTemplateArguments(methodSymbol.GetAttributes())));
+                        }
+                    }, MethodSyntaxKind);
+
+                    ReportIssues(symbolStartContext, symbol, controllerActionInfo);
+                }
+            }, SymbolKind.NamedType);
+        });
+
+    protected bool IsGeneratedCode(SyntaxReference syntaxReference) =>
+        syntaxReference.GetSyntax().SyntaxTree.IsGenerated(GeneratedCodeRecognizer, null);
```

Comment:
The Compilation is `null` because it's not used at all in the method.
I'm not sure why it's there. 
I'll do a cleanup PR to remove it.